portfolio/src/main/java/com/google/sps/servlets/CommentProcessorServlet.java

```diff
@@ -0,0 +1,46 @@
+package com.google.sps.servlets;
+
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import com.google.gson.Gson;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Arrays;
+
+/** Servlet that processes comment. TODO: Improve formatting of response output*/
+@WebServlet("/comment")
+public final class CommentProcessorServlet extends HttpServlet {
+
+  private final List<String> commentList = new ArrayList<String>();
+  private final Gson gson = new Gson(); 
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // Get the input from the form.
+    String comment = getParameter(request, "comment-input", "");
+    commentList.add(comment);
+    response.sendRedirect("/");
+  }
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String json = gson.toJson(commentList);
+    response.setContentType("text/html;");
+    response.getWriter().println(json);
+  }
+
+  /**
+   * @return the request parameter, or the default value if the parameter
+   *         was not specified by the client
+   */
+  private String getParameter(HttpServletRequest request, String name, String defaultValue) {
+    String value = request.getParameter(name);
```

Comment:
What's the reasoning behind the defaultValue? E.g, if we receive a post request for an empty comment, do we still proceed with adding this empty comment?

Reply:
I would think that defaultValue is just defensive programming if something sends a POST request that is not a "comment-input". Not sure if this counts as defensive or unnecessary coding tbh.

Reply:
Yeah I agree that we should be defensive on programming. But it doesn't quite make sense (at least to me) to add an empty comment to our datastore. To be defensive, we can still have check on whether or not the comment is null/empty, but choose to no-op in such case. Or return an errro message in the response. wdyt?

Reply:
Oh. so i checked the getParameter [https://docs.oracle.com/javaee/6/api/javax/servlet/ServletRequest.html#getParameter(java.lang.String)](url)
Returns the value of a request parameter as a String, or **null if the parameter does not exist**. 

I see then in this case, I do not think my getParameter method is necessary, and I can just make the getParameter within the doPost method.

Do let me know if I did that right, thank you

Reply:
Yep make sense, thanks for checking. How do you plan to handle the case if we get a null from parameter?

Reply:
i'm going to do this
`    if (comment != null && !comment.isEmpty()) {
       commentList.add(comment); 
    }`

(it's going to be in my next commit that I'll push now)

Reply:
thank you for

> But it doesn't quite make sense (at least to me) to add an empty comment to our datastore

thank you for challenging my thinking, it really helps

Reply:
Sounds good thanks!